djangoProject_dz3/urls.py

```diff
@@ -15,7 +15,47 @@
 """
 from django.contrib import admin
 from django.urls import path
+from django.http import HttpRequest, HttpResponse
+import random
+
+
+def home(request: HttpRequest) -> HttpResponse:
+    return HttpResponse('Homepage')
+
+def article_info(request: HttpRequest, article_id: int, article_slug: str) -> HttpResponse:
+    try:
+        int(article_id)
+    except:
+        return HttpResponse(f'article_id должен быть только int')
+    return HttpResponse(f'Номер статьи №{article_id}, название статьи "{article_slug}"')
+
+def password_examination(request: HttpRequest, password: str) -> HttpResponse:
+    correct_data = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890'
```

Comment:
Тут не має літер у нижньому регистрі
```suggestion
    valid_characters = string.ascii_letters + string.digits
```